vm/systemSmartContracts/delegation.go

```diff
@@ -255,13 +272,15 @@ func (d *delegation) init(args *vmcommon.ContractCallInput) vmcommon.ReturnCode
 		NotStakedKeys: make([]*NodesData, 0),
 		UnStakedKeys:  make([]*NodesData, 0),
 	}
+	//TODO: If saveDelegationStatus will fail, then the last saves done will remain saved. Is this ok?
 	err = d.saveDelegationStatus(dStatus)
 	if err != nil {
 		d.eei.AddReturnMessage(err.Error())
 		return vmcommon.UserError
 	}
 
 	var fundKey []byte
+	//TODO: If createAndSaveNextFund will fail, then the last saves done will remain saved. Is this ok?
```

Comment:
nothing will remain. Any error invalidates all previous changes. save is done in a cache - cache is invalidated if returnCode != vmcommon.Ok

Reply:
ok